mmengine/runner/runner.py

```diff
@@ -2098,14 +2102,28 @@ def save_checkpoint(
         if hasattr(self.train_dataloader.dataset, 'metainfo'):
             meta.update(dataset_meta=self.train_dataloader.dataset.metainfo)
 
-        if is_model_wrapper(self.model):
-            model = self.model.module
+        # models with this attribute are FSDP models, which has special logic
+        # for `state_dict()` and should run across all processes
+        if hasattr(self.model, '_collect_state_dict'):
+            state_dict = self.model._collect_state_dict()
```

Comment:
Calling a private method of the model wrapper is not safe.

Reply:
I'll rename it to a public method. This is a private method that is only defined in our `MMFSDP` class. It is used to simplify code in `runner`.